src/Tizen.Applications.ComponentManager/Tizen.Applications/ComponentManager.cs

```diff
@@ -0,0 +1,186 @@
+        /// Gets the information of the installed components asynchronously.
+        /// </summary>
+        /// <returns>The installed component info list.</returns>
+        /// <exception cref="ArgumentException">Thrown when failed because of an invalid argument.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed because of the "component not exist" error or the system error.</exception>
+        /// <exception cref="OutOfMemoryException">Thrown when failed because of ouf of memory.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when failed because of permission denied.</exception>
+        /// <privilege>http://tizen.org/privilege/packagemanager.info</privilege>
+        /// <since_tizen> 6 </since_tizen>
+        public static async Task<IEnumerable<ComponentInfo>> GetInstalledComponentsAsync()
+        {
+            return await Task.Run(() =>
+            {
+                Interop.ComponentManager.ErrorCode err = Interop.ComponentManager.ErrorCode.None;
+                List<ComponentInfo> result = new List<ComponentInfo>();
+
+                Interop.ComponentManager.ComponentManagerComponentInfoCallback cb = (IntPtr infoHandle, IntPtr userData) =>
+                {
+                    if (infoHandle != IntPtr.Zero)
+                    {
+                        IntPtr clonedHandle = IntPtr.Zero;
+                        err = Interop.ComponentManager.ComponentInfoClone(out clonedHandle, infoHandle);
+                        if (err != Interop.ComponentManager.ErrorCode.None)
+                        {
+                            Log.Warn(LogTag, "Failed to clone the ComponentInfo. err = " + err);
+                            return false;
+                        }
+                        ComponentInfo info = new ComponentInfo(clonedHandle);
+                        result.Add(info);
+                        return true;
+                    }
+                    return false;
+                };
+                err = Interop.ComponentManager.ComponentManagerForeachComponentInfo(cb, IntPtr.Zero);
+                if (err != Interop.ComponentManager.ErrorCode.None)
+                {
+                    throw ComponentManagerErrorFactory.GetException(err, "Failed to retrieve the component info.");
+                }
+                return result;
+            }).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Gets the information of the running components asynchronously.
+        /// </summary>
+        /// <returns>The component running context list.</returns>
+        /// <exception cref="ArgumentException">Thrown when failed because of an invalid argument.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed because of the "component not exist" error or the system error.</exception>
+        /// <exception cref="OutOfMemoryException">Thrown when failed because of ouf of memory.</exception>
```

Comment:
Spelling, should be `out` instead of `ouf`.

Reply:
Done